SDK_versioned_docs/version-3.0.0/subgraph-example.md

```diff
@@ -0,0 +1,282 @@
+
+This query fetches data about the sender, receiver, amounts, transaction data, and timestamp for a particular swap.
+```
+{
+   swap(id: "0x000007e1111cbd97f74cfc6eea2879a5b02020f26960ac06f4af0f9395372b64#66785") {
+    sender
+    recipient
+    amount0
+    amount1
+    transaction {
+      id
+      blockNumber
+      gasUsed
+      gasPrice
+    }
+    timestamp
+    token0 {
+      id
+      symbol
+    }
+    token1 {
+      id
+      symbol
+    }
+   }
+ }
+ ```
+
+### Recent Swaps Within a Pool
+You can also filter swap data by a pool address by setting the `where` field. This example fetches data about multiple swaps for the USDC-USDT pool, ordered by timestamp.
+
+```
+{
+swaps(orderBy: timestamp, orderDirection: desc, where:
+ { pool: "0x7858e59e0c01ea06df3af3d20ac7b0003275d4bf" }
+) {
+  pool {
+    token0 {
+      id
+      symbol
+    }
+    token1 {
+      id
+      symbol
+    }
+  }
+  sender
+  recipient
+  amount0
+  amount1
+ }
+}
+```
+## Token Data
+
+Fetch token data by inputting the token contract address. Any token that is apart of a Uniswap pool can be queried and the data that is returned is aggregated across all the pools the token is apart of.
+
+### General Token Data
+
+This queries the decimals, symbol, name, pool count, and volume in USD for the UNI token. Reference the full [token schema](https://github.com/Uniswap/v3-subgraph/blob/main/schema.graphql#L38) for all possible fields you can query.
+
+
+```
+{
+  token(id:"0x1f9840a85d5af5bf1d1762f925bdaddc4201f984") {
+    symbol
+    name
+    decimals
+    volumeUSD
+    poolCount
+  }
+}
+```
+
+### Token Daily Aggregated
+
+You can fetch aggregate data about a specific token over a day. This query gets 10-day volume data for the UNI token ordered from oldest to newest.
+```
+{
+  tokenDayDatas(first: 10, where: {token: "0x1f9840a85d5af5bf1d1762f925bdaddc4201f984"}, orderBy: date, orderDirection: asc) {
+    date
+    token {
+      id
+      symbol
+    }
+    volumeUSD
+  }
+}
+```
+
+### All Tokens
+
+Similar to retrieving all pools, you can fetch all tokens by using skip.
+Note: This query will not work in the graph sandbox and more resembles the structure of a query you'd pass to some graphql middleware like Apollo.
+
+```
+query tokens($skip: Int!) {
+  tokens(first: 1000, skip: $skip) {
+    id
+    symbol
+    name
+  }
+}
+```
+
+## Position Data
+
+### General Position Data
+To get data about a specific position, input the NFT tokenId. This queries the collected fees for token0 and token1 and current liquidity for the position with tokedId 3. Reference the full [position schema](https://github.com/Uniswap/v3-subgraph/blob/main/schema.graphql#L192) to see all fields.
+
+```
+{
+  position(id:3) {
+    id
+    collectedFeesToken0
+    collectedFeesToken1
+    liquidity
+    token0 {
+      id
+      symbol
+    }
+    token1
+    {
+      id
+      symbol
+    }
+  }
+}
```

Comment:
is it possible to apply color formatting to the code examples? readability would be much improved if keywords, variable names, etc were color coded

Reply:
from what I can tell markdown doesnt support graphql code blocks :(